# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpBackgroundChannelChild.cpp
# Commit: be03e0e7b645
# Full Hash: be03e0e7b64568618b834834602b5ba303b32214
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-24 09:31:07
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P3 send OS*R and ODA via pHttpChannel for multipart channel, r=mayhemer,necko-reviewers
#   
#   Depends on D76969
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76970
# ==============================================================================

diff -r df5abc2c0734 -r be03e0e7b645 netwerk/protocol/http/HttpBackgroundChannelChild.cpp
--- a/netwerk/protocol/http/HttpBackgroundChannelChild.cpp	Tue Jun 23 16:56:32 2020 +0000
+++ b/netwerk/protocol/http/HttpBackgroundChannelChild.cpp	Tue Jun 23 16:56:45 2020 +0000
@@ -83,11 +83,12 @@
   return !mChannelChild;
 }
 
-void HttpBackgroundChannelChild::OnStartRequestReceived() {
+void HttpBackgroundChannelChild::OnStartRequestReceived(
+    Maybe<uint32_t> aMultiPartID) {
   LOG(("HttpBackgroundChannelChild::OnStartRequestReceived [this=%p]\n", this));
   MOZ_ASSERT(OnSocketThread());
   MOZ_ASSERT(mChannelChild);
-  MOZ_ASSERT(!mStartReceived);  // Should only be called once.
+  MOZ_ASSERT(!mStartReceived || *aMultiPartID > 0);
 
   mStartReceived = true;
 
@@ -149,7 +150,7 @@
                                        aRequestHeaders, aArgs);
   // Allow to queue other runnable since OnStartRequest Event already hits the
   // child's mEventQ.
-  OnStartRequestReceived();
+  OnStartRequestReceived(aArgs.multiPartID());
 
   return IPC_OK();
 }